src/api/data/cache/cache.go

```diff
@@ -148,37 +148,84 @@ func (c *cachedCharts) Refresh() error {
 		if err != nil {
 			return err
 		}
+
+		// 2 - Parse repo index
 		charts, err := helpers.ParseYAMLRepo(body, repo.Name)
 		if err != nil {
 			return err
 		}
+
+		// 3 - Process elements in index
 		var chartsWithData []*models.ChartPackage
+		// Buffered channel
+		ch := make(chan chanItem, len(charts))
```

Comment:
Using buffered channels to avoid deadlocks and routine leak in case of error in the drain procedure.